Sources/CoreMethods/Public/UI/Fields/CardNumber/CardNumberTextField.swift

```
@@ -0,0 +1,239 @@
+    /// - Note: BIN consists of the first 8 digits of the card number.
+    /// - Parameter bin: The current BIN value
+    public var onBinChanged: ((String) -> Void)?
+
+    /// Callback triggered when a valid card number is completed.
+    /// - Parameter lastFour: The last four digits of the validated card number
+    public var onLastFourDigitsFilled: ((String) -> Void)?
+
+    /// Callback triggered when the field focus state changes.
+    /// - Parameter isFocused: True when field gains focus, false when it loses focus
+    public var onFocusChanged: ((Bool) -> Void)?
+
+    ///
+    /// Callback triggered when a validation error occurs.
+    ///
+    /// # This callback is triggered in two scenarios
+    /// * When the field loses focus and contains an invalid card number.
+    /// * When the maximum length is reached but validation fails.
+    /// - Parameter error: The type of validation error that occurred
+    public var onError: ((CardNumberError) -> Void)?
+
+    /// Returns whether the current input represents a valid card number.
+    public var isValid: Bool {
+        return self.input.isValid
+    }
+
+    /// The current number of digits entered in the field.
+    public var count: Int {
+        return self.input.count
+    }
+
+    // MARK: - Input Configuration
+
+    /// A Boolean value indicating whether the field is enabled
+    public var isEnabled: Bool {
+        get { self.input.isEnabled }
+        set { self.input.isEnabled = newValue }
+    }
+
+    /// The appearance of the keyboard that is associated with the text field
+    public var keyboardAppearance: UIKeyboardAppearance {
+        get { self.input.keyboardAppearance }
+        set { self.input.keyboardAppearance = newValue }
+    }
+
+    private let validation = CardNumberValidation()
+
+    private let binLength = 8
+
+    let input: PCIFieldState
+
+    // MARK: - Initialization
+
+    public init(style: Style = TextFieldDefaultStyle(), maxLength: Int = 16) {
```

there are cases in which the length can be 19 so it would be good if the max here is 19

---

Card numbers with 19 digits are rare compared to 16-digit numbers. Wouldn't it be better to have 16 as the default and set the mask accordingly?

---

The Core Methods SDK JS has a maximum length of 16 as default, and the length needs to be between 8 and 19.

https://beta-sdk.mercadopago.com/js/v2/demo/prod/fieldsCoreMethods.html
https://github.com/mercadopago/sdk-js/blob/main/docs/fields.md#fields-module

---

@icaldana @meliguicarvalho Is that right ?

---

my only concern is if the integrator does not change it and a user with a card with 19 digits cannot type it. we can also discuss this and change it in the future if necessary not to block this pull request

---

I understand your concern. We can discuss this further later.